pkg/netstore/netstore.go

```diff
@@ -34,11 +42,23 @@ func (s *store) Get(ctx context.Context, mode storage.ModeGet, addr swarm.Addres
 	if err != nil {
 		if errors.Is(err, storage.ErrNotFound) {
 			// request from network
-			ch, err := s.retrieval.RetrieveChunk(ctx, addr)
+			ch, err = s.retrieval.RetrieveChunk(ctx, addr)
 			if err != nil {
-				return nil, fmt.Errorf("netstore retrieve chunk: %w", err)
+				if s.recoveryCallback == nil {
+					return nil, err
+				}
+				targets, err := sctx.GetTargets(ctx)
+				if err != nil {
+					return nil, err
+				}
+				go func() {
```

Comment:
Related to my comments on the context, if recovery.RecoveryHook would accept a context argument, it would be possible to terminate this goroutine when netstore is closed, or maybe to set it a timeout. I think that it is important to control the lifetime of our goroutines.

Reply:
I am not taking this for now in this PR as the plan is to have a "instant repair" feature, which means we have to block in netstore until the response comes or timeouts.